colosseum/parser.py

```diff
@@ -126,3 +128,133 @@ def color(value):
+            LINE_HEIGHT_CHOICES.validate(font_dict['line_height'])
+            return font_dict, True
+        else:
+            # Or just a font size
+            try:
+                FONT_SIZE_CHOICES.validate(value)
+                font_dict['font_size'] = value
+                return font_dict, True
+            except ValueError:
+                pass
+
+        raise ValidationError('Font value "{value}" not valid!'.format(value=value))
+
+    return font_dict, False
+
+
+def parse_font(string):
+    """
+    Parse font string into a dictionary of font properties.
+
+    The font CSS property is a shorthand for font-style, font-variant, font-weight,
+    font-size, line-height, and font-family.
+
+    Alternatively, it sets an element's font to a system font.
+
+    Reference:
+    - https://www.w3.org/TR/CSS21/fonts.html#font-shorthand
+    - https://developer.mozilla.org/en-US/docs/Web/CSS/font
+    """
+    from .constants import INHERIT, INITIAL_FONT_VALUES, SYSTEM_FONT_KEYWORDS, FONT_FAMILY_CHOICES  # noqa
+    font_dict = INITIAL_FONT_VALUES.copy()
+
+    # Remove extra spaces
+    string = ' '.join(str(string).strip().split())
+    parts = string.split(' ', 1)
+    if len(parts) == 1:
+        # If font is specified as a system keyword, it must be one of:
+        # caption, icon, menu, message-box, small-caption, status-bar
+        value = parts[0]
+        if value == INHERIT:
+            # TODO: To be completed by future work
+            pass
+        else:
+            if value not in SYSTEM_FONT_KEYWORDS:
+                error_msg = ('Font property value "{value}" '
+                             'not a system font keyword!'.format(value=value))
+                raise ValueError(error_msg)
+            font_dict = get_system_font(value)
+    else:
+        # If font is specified as a shorthand for several font-related properties, then:
+        #  - It must include values for:
+        #    <font-size> and <font-family>
+        #  - It may optionally include values for:
+        #    <font-style> <font-variant> <font-weight> and <line-height>
+        #  - font-style, font-variant and font-weight must precede font-size
+        #  - font-variant may only specify the values defined in CSS 2.1
+        #  - line-height must immediately follow font-size, preceded by "/", like this: "16px/3"
+        #  - font-family must be the last value specified.
+
+        # We iteratively split by the first left hand space found and try to validate if that part
+        # is a valid <font-style> or <font-variant> or <font-weight> (which can come in any order)
+        # or <font-size>/<line-height> (which has to come after all the other properties)
+        old_is_font_size = False  # Need to check that some properties come after font-size
+        for _ in range(5):
+            value = parts[0]
+            try:
+                font_dict, is_font_size = _parse_font_property_part(value, font_dict)
```

Comment:
Hmmm interesting. I think this will 💥 😆

Reply:
>What happens here in the case of "normal"? _parse_font_property_part ignores normal, but I can't see how that property is picked up if it's manually specified for style or variant?

I skip `normal` during the parsing process because it brings no information regarding which property it refers to: is it a font_style? is it a font_variant? is it a font_weight?

So values that do provide information on these are used. Now since we start with a dictionary of initial values (where all those properties are in facto normal) if the user used normal, then that will be reflected at the end anyway.

>Also, what happens if you specify something daft like "bigger smaller Helvetica" - i.e., double specify one of properties with entirely legal, but contradictory values?

Added a check for this! nice catch